### PR TITLE
[LoongArch] Support LA V1.1 feature ld-seq-sa that don't generate dbar 0x700.

### PR DESCRIPTION
**Pull request by tangaac**
Two options for clang
  -mld-seq-sa:             Do not generate load-load barrier instructions (dbar 0x700)
  -mno-ld-seq-sa:          Generate load-load barrier instructions (dbar 0x700)
The default is -mno-ld-seq-sa

### CODE REVIEW

File: llvm/test/CodeGen/LoongArch/ir-instruction/atomic-cmpxchg.ll [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
 ; RUN: llc --mtriple=loongarch64 -mattr=+d < %s | FileCheck %s --check-prefix=LA64
+; RUN: llc --mtriple=loongarch64 -mattr=+d,+ld-seq-sa < %s | FileCheck %s --check-prefix=LA64-LD-SEQ-SA
```
Comment (SixWeining) [inline llvm/test/CodeGen/LoongArch/ir-instruction/atomic-cmpxchg.ll:3]: To reduce test lines:
```
; RUN: llc --mtriple=loongarch64 -mattr=+d,-ld-seq-sa < %s | FileCheck %s --check-prefixes=LA64,NO-LD-SEQ-SA
; RUN: llc --mtriple=loongarch64 -mattr=+d,+ld-seq-sa < %s | FileCheck %s --check-prefix=LA64
```

File: llvm/test/CodeGen/LoongArch/ir-instruction/atomic-cmpxchg.ll [hunk 2]
```diff
@@ -100,99 +101,99 @@ define void @cmpxchg_i64_acquire_acquire(ptr %ptr, i64 %cmp, i64 %val) nounwind
 }
 
 define void @cmpxchg_i8_acquire_monotonic(ptr %ptr, i8 %cmp, i8 %val) nounwind {
```
Comment (SixWeining) [inline llvm/test/CodeGen/LoongArch/ir-instruction/atomic-cmpxchg.ll:103]: This test doesn't have `LA64` check?

**Review by SixWeining — APPROVED** [review]
LGTM. But we can also add this feature in `sys::getHostCPUFeatures` in this PR.

File: llvm/lib/TargetParser/Host.cpp [hunk 3]
```diff
@@ -2011,8 +2011,9 @@ const StringMap<bool> sys::getHostCPUFeatures() {
 const StringMap<bool> sys::getHostCPUFeatures() {
   unsigned long hwcap = getauxval(AT_HWCAP);
   bool HasFPU = hwcap & (1UL << 3); // HWCAP_LOONGARCH_FPU
-  uint32_t cpucfg2 = 0x2;
+  uint32_t cpucfg2 = 0x2, cpucfg3 = 0x3;
```
Comment (SixWeining) [inline llvm/lib/TargetParser/Host.cpp:2014]:
```suggestion
  const uint32_t cpucfg2 = 0x2, cpucfg3 = 0x3;
```

**Review by SixWeining — APPROVED** [review]
LGTM except a nit.

**Review by SixWeining — APPROVED** [review]

File: llvm/lib/TargetParser/Host.cpp [hunk 4]
```diff
@@ -2011,8 +2011,9 @@ const StringMap<bool> sys::getHostCPUFeatures() {
 const StringMap<bool> sys::getHostCPUFeatures() {
   unsigned long hwcap = getauxval(AT_HWCAP);
   bool HasFPU = hwcap & (1UL << 3); // HWCAP_LOONGARCH_FPU
-  uint32_t cpucfg2 = 0x2;
+  const uint32_t cpucfg2 = 0x2, cpucfg3 = 0x3;
```
Comment (dtcxzyw) [inline llvm/lib/TargetParser/Host.cpp:2014]: I encountered a compilation error with gcc 13.2:
```
[1/371] Building CXX object lib/TargetParser/CMakeFiles/LLVMTargetParser.dir/Host.cpp.o
FAILED: lib/TargetParser/CMakeFiles/LLVMTargetParser.dir/Host.cpp.o 
ccache /usr/bin/g++ -DGTEST_HAS_RTTI=0 -DLLVM_EXPORTS -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/dtcxzyw/llvm-build/lib/TargetParser -I/home/dtcxzyw/llvm-project/llvm/lib/TargetParser -I/home/dtcxzyw/llvm-build/include -I/home/dtcxzyw/llvm-project/llvm/include -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -fno-lifetime-dse -w -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG -std=c++17 -fPIC  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -MD -MT lib/TargetParser/CMakeFiles/LLVMTargetParser.dir/Host.cpp.o -MF lib/TargetParser/CMakeFiles/LLVMTargetParser.dir/Host.cpp.o.d -o lib/TargetParser/CMakeFiles/LLVMTargetParser.dir/Host.cpp.o -c /home/dtcxzyw/llvm-project/llvm/lib/TargetParser/Host.cpp
/home/dtcxzyw/llvm-project/llvm/lib/TargetParser/Host.cpp: In function ‘const llvm::StringMap<bool, llvm::MallocAllocator> llvm::sys::getHostCPUFeatures()’:
/home/dtcxzyw/llvm-project/llvm/lib/TargetParser/Host.cpp:2015:3: error: read-only variable ‘cpucfg2’ used as ‘asm’ output
 2015 |   __asm__("cpucfg %[cpucfg2], %[cpucfg2]\n\t" : [cpucfg2] "+r"(cpucfg2));
      |   ^~~~~~~
/home/dtcxzyw/llvm-project/llvm/lib/TargetParser/Host.cpp:2016:3: error: read-only variable ‘cpucfg3’ used as ‘asm’ output
 2016 |   __asm__("cpucfg %[cpucfg3], %[cpucfg3]\n\t" : [cpucfg3] "+r"(cpucfg3));
      |   ^~~~~~~
ninja: build stopped: subcommand failed.
```

Comment (SixWeining) [reply llvm/lib/TargetParser/Host.cpp:2014]: Should already be fixed.

**Pull request merged**